### PR TITLE
Fix the deprecating issue in the query operation

### PR DESCRIPTION
**Pull request by sachinira**
# Description
$subject

Related Pull Requests
- https://github.com/ballerina-platform/module-ballerinax-sfdc/pull/251

One line release note: 
- Fix the connector by deprecating the `getQueryResult` and `searchSOSLString` operations
- Introduce `getQueryResultStream` and `searchSOSLStringStream`  for the `getQueryResult` and `searchSOSLString` operations respectively

## Type of change

Please delete options that are not relevant.
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

**Test Configuration**:
* Ballerina Version: 2201.0.2
* Operating System:
* Java SDK: 

# Checklist:

### Security checks
 - [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: salesforce/samples/rest_api_usecases/find_record_by_query_stream.bal [hunk 1]
```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (LakshanSS) [inline salesforce/samples/rest_api_usecases/find_record_by_query_stream.bal:1]:
```suggestion
// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

**Comment by SkNuwanTissera** [comment]
@sachinira Seems like these 2 templates are affected. Let's fix them after this 
image

**Comment by SkNuwanTissera** [comment]
> @sachinira Seems like these 2 templates are affected. Let's fix them after this image

Since this is a deprecation. Do we need to change?. WDYT?

**Comment by sachinira** [comment]
> > @sachinira Seems like these 2 templates are affected. Let's fix them after this image
> 
> Since this is a deprecation. Do we need to change?. WDYT?

Yes we need to update

**Review by SkNuwanTissera — APPROVED** [review]

**Review by LakshanSS — APPROVED** [review]

**Pull request merged**